src/components/EnterpriseApp/index.jsx

```diff
@@ -125,6 +126,10 @@ class EnterpriseApp extends React.Component {
       return <EnterpriseAppSkeleton />;
     }
 
+    if (!loading && enterpriseId === null) {
```

Comment:
nit: we can assume `loading` is false if we're getting to this conditional due to the conditional on line 125, so its unnecessary to check on this line :)

curious: it may also make sense to use `if (!enterpriseId) { ... }` rather than specifically checking for `null` to catch cases where it may be `undefined` or `null` (in both situations, we'd want to display the `NotFoundPage`.